web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/SelectSecurityGroups.tsx

```diff
@@ -268,18 +270,21 @@ function getTrustedSecurityGroups(
   securityGroups: SecurityGroup[],
   db: AwsRdsDatabase
 ): Set<string> {
-  const securityGroupsById = new Map(
-    securityGroups.map(group => [group.id, group])
-  );
+  const trustedGroups = new Set<string>();
+  if (!db) {
+    return trustedGroups;
+  }
 
   const dbPort = getPort(db);
-  const trustedGroups = new Set<string>();
-  db.securityGroups.forEach(groupId => {
+  const securityGroupsById = new Map(
+    securityGroups?.map(group => [group.id, group])
+  );
+  db?.securityGroups?.forEach(groupId => {
     const group = securityGroupsById.get(groupId);
     if (!group) {
       return;
```

Comment:
here too, the return type should be consistent,(our typescript setting isn't perfect)

return type `Set<string> | undefined`, and handle the undefined case at the call site

Reply:
Why make it undefined? If we can't determine the trusted security groups then an empty set works... I didn't change the return type

Reply:
b/c lets say there was no `group`, the callsite expects a set, but it returns undefined there (line 285), but i dont think we will have a `empty group` case will we though?

Reply:
oh, it returns inside the `forEach` callback func. The set itself will be defined even if there are no security groups